packages/lit-element/src/lib/decorators/customElement.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/*
+ * IMPORTANT: For compatibility with tsickle and the Closure JS compiler, all
+ * property decorators (but not class decorators) in this file that have
+ * an @ExportDecoratedItems annotation must be defined as a regular function,
+ * not an arrow function.
+ */
+import {Constructor, ClassDescriptor} from './base.js';
+
+const legacyCustomElement = (
+  tagName: string,
+  clazz: Constructor<HTMLElement>
+) => {
+  window.customElements.define(tagName, clazz);
+  // Cast as any because TS doesn't recognize the return type as being a
+  // subtype of the decorated class when clazz is typed as
+  // `Constructor<HTMLElement>` for some reason.
+  // `Constructor<HTMLElement>` is helpful to make sure the decorator is
+  // applied to elements however.
+  // tslint:disable-next-line:no-any
+  return clazz as any;
+};
+
+const standardCustomElement = (
+  tagName: string,
+  descriptor: ClassDescriptor
+) => {
+  const {kind, elements} = descriptor;
+  return {
+    kind,
+    elements,
+    // This callback is called once the class is otherwise fully defined
+    finisher(clazz: Constructor<HTMLElement>) {
+      window.customElements.define(tagName, clazz);
+    },
+  };
+};
+
+/**
+ * Class decorator factory that defines the decorated class as a custom element.
+ *
+ * ```
+ * @customElement('my-element')
+ * class MyElement {
+ *   render() {
+ *     return html``;
+ *   }
+ * }
+ * ```
+ * @category Decorator
+ * @param tagName The name of the custom element to define.
+ */
+export const customElement = (tagName: string) => (
```

Comment:
btw, I'd love to bring back the type here of `tagName: keyof HTMLElementTagNameMap`. There was a problem with that type being expanded to a union at our compile time, but I think there are workarounds, like maybe using a `function` instead of an arrow function.